### PR TITLE
Fix issue with newer MongoDB requiring TLS trust chain

### PR DESCRIPTION
**Pull request by tianon**
See https://jira.mongodb.org/browse/SERVER-72839

```console
$ ./test/run.sh -t mongo-tls-basics -t mongo-tls-auth mongo:4.4.29
testing mongo:4.4.29
	'mongo-tls-basics' [1/2]...librarytest/mongo-tls:mongo-4.4.29 stopped unexpectedly!
++ docker logs c6b35d558bcd95eeb36932e02b289b1d9e62283aa05e143a1cda20b6fbcf476f
{"t":{"$date":"2024-02-28T22:56:30.628Z"},"s":"F",  "c":"CONTROL",  "id":20574,   "ctx":"main","msg":"Error during global initialization","attr":{"error":{"code":72,"codeName":"InvalidOptions","errmsg":"The use of TLS without specifying a chain of trust is no longer supported. See https://jira.mongodb.org/browse/SERVER-72839 for details."}}}
failed
	'mongo-tls-auth' [2/2]...librarytest/mongo-tls:mongo-4.4.29 stopped unexpectedly!
++ docker logs 0417420c857d936cbd80e34f43450b624ada55e543de3fa42325d7a7fe7876fc
{"t":{"$date":"2024-02-28T22:56:32.344Z"},"s":"F",  "c":"CONTROL",  "id":20574,   "ctx":"main","msg":"Error during global initialization","attr":{"error":{"code":72,"codeName":"InvalidOptions","errmsg":"The use of TLS without specifying a chain of trust is no longer supported. See https://jira.mongodb.org/browse/SERVER-72839 for details."}}}
failed
```

### CODE REVIEW

**Comment by tianon** [comment]
Marking as a draft because while this _should_ be enough to fix it, it is _not_: :upside_down_face:

```console
$ ./test/run.sh -t mongo-tls-basics mongo:4.4.29
testing mongo:4.4.29
	'mongo-tls-basics' [1/1]............librarytest/mongo-tls:mongo-4.4.29 failed to accept connections in a reasonable amount of time!
++ docker logs 52f1f8093584a1e7c9d458314ab3a2e4355d0f87afc31607d4dbf7cb8aa1cb73
{"t":{"$date":"2024-02-28T23:00:17.809+00:00"},"s":"W",  "c":"NETWORK",  "id":551190,  "ctx":"main","msg":"Server certificate has no compatible Subject Alternative Name. This may prevent TLS clients from connecting","tags":["startupWarnings"]}
{"t":{"$date":"2024-02-28T23:00:17.809+00:00"},"s":"I",  "c":"CONTROL",  "id":23285,   "ctx":"main","msg":"Automatically disabling TLS 1.0, to force-enable TLS 1.0 specify --sslDisabledProtocols 'none'"}
{"t":{"$date":"2024-02-28T23:00:17.811+00:00"},"s":"I",  "c":"NETWORK",  "id":4648601, "ctx":"main","msg":"Implicit TCP FastOpen unavailable. If TCP FastOpen is required, set tcpFastOpenServer, tcpFastOpenClient, and tcpFastOpenQueueSize."}
{"t":{"$date":"2024-02-28T23:00:17.812+00:00"},"s":"I",  "c":"STORAGE",  "id":4615611, "ctx":"initandlisten","msg":"MongoDB starting","attr":{"pid":1,"port":27017,"dbPath":"/data/db","architecture":"64-bit","host":"mongo"}}
{"t":{"$date":"2024-02-28T23:00:17.812+00:00"},"s":"I",  "c":"CONTROL",  "id":23403,   "ctx":"initandlisten","msg":"Build Info","attr":{"buildInfo":{"version":"4.4.29","gitVersion":"f4dda329a99811c707eb06d05ad023599f9be263","openSSLVersion":"OpenSSL 1.1.1f  31 Mar 2020","modules":[],"allocator":"tcmalloc","environment":{"distmod":"ubuntu2004","distarch":"x86_64","target_arch":"x86_64"}}}}
{"t":{"$date":"2024-02-28T23:00:17.812+00:00"},"s":"I",  "c":"CONTROL",  "id":51765,   "ctx":"initandlisten","msg":"Operating System","attr":{"os":{"name":"Ubuntu","version":"20.04"}}}
{"t":{"$date":"2024-02-28T23:00:17.812+00:00"},"s":"I",  "c":"CONTROL",  "id":21951,   "ctx":"initandlisten","msg":"Options set by command line","attr":{"options":{"net":{"bindIp":"*","tls":{"CAFile":"/certs/ca.crt","certificateKeyFile":"/certs/both.pem","mode":"requireTLS"}}}}}
{"t":{"$date":"2024-02-28T23:00:17.813+00:00"},"s":"I",  "c":"STORAGE",  "id":22297,   "ctx":"initandlisten","msg":"Using the XFS filesystem is strongly recommended with the WiredTiger storage engine. See http://dochub.mongodb.org/core/prodnotes-filesystem","tags":["startupWarnings"]}
{"t":{"$date":"2024-02-28T23:00:17.813+00:00"},"s":"I",  "c":"STORAGE",  "id":22315,   "ctx":"initandlisten","msg":"Opening WiredTiger","attr":{"config":"create,cache_size=31453M,session_max=33000,eviction=(threads_min=4,threads_max=4),config_base=false,statistics=(fast),log=(enabled=true,archive=true,path=journal,compressor=snappy),file_manager=(close_idle_time=100000,close_scan_interval=10,close_handle_minimum=250),statistics_log=(wait=0),verbose=[recovery_progress,checkpoint_progress,compact_progress],"}}
{"t":{"$date":"2024-02-28T23:00:18.282+00:00"},"s":"I",  "c":"STORAGE",  "id":22430,   "ctx":"initandlisten","msg":"WiredTiger message","attr":{"message":"[1709161218:282772][1:0x7fc927bd4cc0], txn-recover: [WT_VERB_RECOVERY | WT_VERB_RECOVERY_PROGRESS] Set global recovery timestamp: (0, 0)"}}
{"t":{"$date":"2024-02-28T23:00:18.282+00:00"},"s":"I",  "c":"STORAGE",  "id":22430,   "ctx":"initandlisten","msg":"WiredTiger message","attr":{"message":"[1709161218:282807][1:0x7fc927bd4cc0], txn-recover: [WT_VERB_RECOVERY | WT_VERB_RECOVERY_PROGRESS] Set global oldest timestamp: (0, 0)"}}
{"t":{"$date":"2024-02-28T23:00:18.291+00:00"},"s":"I",  "c":"STORAGE",  "id":4795906, "ctx":"initandlisten","msg":"WiredTiger opened","attr":{"durationMillis":478}}
{"t":{"$date":"2024-02-28T23:00:18.291+00:00"},"s":"I",  "c":"RECOVERY", "id":23987,   "ctx":"initandlisten","msg":"WiredTiger recoveryTimestamp","attr":{"recoveryTimestamp":{"$timestamp":{"t":0,"i":0}}}}
{"t":{"$date":"2024-02-28T23:00:18.305+00:00"},"s":"I",  "c":"STORAGE",  "id":22262,   "ctx":"initandlisten","msg":"Timestamp monitor starting"}
{"t":{"$date":"2024-02-28T23:00:18.310+00:00"},"s":"W",  "c":"CONTROL",  "id":22120,   "ctx":"initandlisten","msg":"Access control is not enabled for the database. Read and write access to data and configuration is unrestricted","tags":["startupWarnings"]}
{"t":{"$date":"2024-02-28T23:00:18.311+00:00"},"s":"W",  "c":"CONTROL",  "id":22178,   "ctx":"initandlisten","msg":"/sys/kernel/mm/transparent_hugepage/enabled is 'always'. We suggest setting it to 'never'","tags":["startupWarnings"]}
{"t":{"$date":"2024-02-28T23:00:18.311+00:00"},"s":"I",  "c":"STORAGE",  "id":20320,   "ctx":"initandlisten","msg":"createCollection","attr":{"namespace":"admin.system.version","uuidDisposition":"provided","uuid":{"uuid":{"$uuid":"9e103be0-fc1e-4cbb-b7ee-471572c4fae3"}},"options":{"uuid":{"$uuid":"9e103be0-fc1e-4cbb-b7ee-471572c4fae3"}}}}
{"t":{"$date":"2024-02-28T23:00:18.322+00:00"},"s":"I",  "c":"INDEX",    "id":20345,   "ctx":"initandlisten","msg":"Index build: done building","attr":{"buildUUID":null,"namespace":"admin.system.version","index":"_id_","commitTimestamp":{"$timestamp":{"t":0,"i":0}}}}
{"t":{"$date":"2024-02-28T23:00:18.322+00:00"},"s":"I",  "c":"COMMAND",  "id":20459,   "ctx":"initandlisten","msg":"Setting featureCompatibilityVersion","attr":{"newVersion":"4.4"}}
{"t":{"$date":"2024-02-28T23:00:18.322+00:00"},"s":"I",  "c":"STORAGE",  "id":20536,   "ctx":"initandlisten","msg":"Flow Control is enabled on this deployment"}
{"t":{"$date":"2024-02-28T23:00:18.323+00:00"},"s":"I",  "c":"STORAGE",  "id":20320,   "ctx":"initandlisten","msg":"createCollection","attr":{"namespace":"local.startup_log","uuidDisposition":"generated","uuid":{"uuid":{"$uuid":"fbf96195-6590-4e22-b0d9-b5678e8db635"}},"options":{"capped":true,"size":10485760}}}
{"t":{"$date":"2024-02-28T23:00:18.334+00:00"},"s":"I",  "c":"INDEX",    "id":20345,   "ctx":"initandlisten","msg":"Index build: done building","attr":{"buildUUID":null,"namespace":"local.startup_log","index":"_id_","commitTimestamp":{"$timestamp":{"t":0,"i":0}}}}
{"t":{"$date":"2024-02-28T23:00:18.334+00:00"},"s":"I",  "c":"FTDC",     "id":20625,   "ctx":"initandlisten","msg":"Initializing full-time diagnostic data capture","attr":{"dataDirectory":"/data/db/diagnostic.data"}}
{"t":{"$date":"2024-02-28T23:00:18.334+00:00"},"s":"I",  "c":"REPL",     "id":6015317, "ctx":"initandlisten","msg":"Setting new configuration state","attr":{"newState":"ConfigReplicationDisabled","oldState":"ConfigPreStart"}}
{"t":{"$date":"2024-02-28T23:00:18.335+00:00"},"s":"I",  "c":"CONTROL",  "id":20712,   "ctx":"LogicalSessionCacheReap","msg":"Sessions collection is not set up; waiting until next sessions reap interval","attr":{"error":"NamespaceNotFound: config.system.sessions does not exist"}}
{"t":{"$date":"2024-02-28T23:00:18.335+00:00"},"s":"I",  "c":"STORAGE",  "id":20320,   "ctx":"LogicalSessionCacheRefresh","msg":"createCollection","attr":{"namespace":"config.system.sessions","uuidDisposition":"generated","uuid":{"uuid":{"$uuid":"3f468379-36f6-461f-bba7-4765ff5674d3"}},"options":{}}}
{"t":{"$date":"2024-02-28T23:00:18.335+00:00"},"s":"I",  "c":"NETWORK",  "id":23015,   "ctx":"listener","msg":"Listening on","attr":{"address":"/tmp/mongodb-27017.sock"}}
{"t":{"$date":"2024-02-28T23:00:18.335+00:00"},"s":"I",  "c":"NETWORK",  "id":23015,   "ctx":"listener","msg":"Listening on","attr":{"address":"0.0.0.0"}}
{"t":{"$date":"2024-02-28T23:00:18.335+00:00"},"s":"I",  "c":"NETWORK",  "id":23016,   "ctx":"listener","msg":"Waiting for connections","attr":{"port":27017,"ssl":"on"}}
{"t":{"$date":"2024-02-28T23:00:18.352+00:00"},"s":"I",  "c":"INDEX",    "id":20345,   "ctx":"LogicalSessionCacheRefresh","msg":"Index build: done building","attr":{"buildUUID":null,"namespace":"config.system.sessions","index":"_id_","commitTimestamp":{"$timestamp":{"t":0,"i":0}}}}
{"t":{"$date":"2024-02-28T23:00:18.352+00:00"},"s":"I",  "c":"INDEX",    "id":20345,   "ctx":"LogicalSessionCacheRefresh","msg":"Index build: done building","attr":{"buildUUID":null,"namespace":"config.system.sessions","index":"lsidTTLIndex","commitTimestamp":{"$timestamp":{"t":0,"i":0}}}}
{"t":{"$date":"2024-02-28T23:00:20.551+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"172.27.0.7:41926","connectionId":1,"connectionCount":1}}
{"t":{"$date":"2024-02-28T23:00:20.558+00:00"},"s":"E",  "c":"NETWORK",  "id":23255,   "ctx":"conn1","msg":"No SSL certificate provided by peer; connection rejected"}
{"t":{"$date":"2024-02-28T23:00:20.558+00:00"},"s":"I",  "c":"NETWORK",  "id":22988,   "ctx":"conn1","msg":"Error receiving request from client. Ending connection from remote","attr":{"error":{"code":141,"codeName":"SSLHandshakeFailed","errmsg":"no SSL certificate provided by peer; connection rejected"},"remote":"172.27.0.7:41926","connectionId":1}}
{"t":{"$date":"2024-02-28T23:00:20.558+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn1","msg":"Connection ended","attr":{"remote":"172.27.0.7:41926","connectionId":1,"connectionCount":0}}
{"t":{"$date":"2024-02-28T23:00:23.128+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"172.27.0.7:41940","connectionId":2,"connectionCount":1}}
{"t":{"$date":"2024-02-28T23:00:23.134+00:00"},"s":"E",  "c":"NETWORK",  "id":23255,   "ctx":"conn2","msg":"No SSL certificate provided by peer; connection rejected"}
{"t":{"$date":"2024-02-28T23:00:23.134+00:00"},"s":"I",  "c":"NETWORK",  "id":22988,   "ctx":"conn2","msg":"Error receiving request from client. Ending connection from remote","attr":{"error":{"code":141,"codeName":"SSLHandshakeFailed","errmsg":"no SSL certificate provided by peer; connection rejected"},"remote":"172.27.0.7:41940","connectionId":2}}
{"t":{"$date":"2024-02-28T23:00:23.134+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn2","msg":"Connection ended","attr":{"remote":"172.27.0.7:41940","connectionId":2,"connectionCount":0}}
{"t":{"$date":"2024-02-28T23:00:25.722+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"172.27.0.7:41946","connectionId":3,"connectionCount":1}}
{"t":{"$date":"2024-02-28T23:00:25.728+00:00"},"s":"E",  "c":"NETWORK",  "id":23255,   "ctx":"conn3","msg":"No SSL certificate provided by peer; connection rejected"}
{"t":{"$date":"2024-02-28T23:00:25.728+00:00"},"s":"I",  "c":"NETWORK",  "id":22988,   "ctx":"conn3","msg":"Error receiving request from client. Ending connection from remote","attr":{"error":{"code":141,"codeName":"SSLHandshakeFailed","errmsg":"no SSL certificate provided by peer; connection rejected"},"remote":"172.27.0.7:41946","connectionId":3}}
{"t":{"$date":"2024-02-28T23:00:25.728+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn3","msg":"Connection ended","attr":{"remote":"172.27.0.7:41946","connectionId":3,"connectionCount":0}}
{"t":{"$date":"2024-02-28T23:00:28.310+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"172.27.0.7:41958","connectionId":4,"connectionCount":1}}
{"t":{"$date":"2024-02-28T23:00:28.316+00:00"},"s":"E",  "c":"NETWORK",  "id":23255,   "ctx":"conn4","msg":"No SSL certificate provided by peer; connection rejected"}
{"t":{"$date":"2024-02-28T23:00:28.316+00:00"},"s":"I",  "c":"NETWORK",  "id":22988,   "ctx":"conn4","msg":"Error receiving request from client. Ending connection from remote","attr":{"error":{"code":141,"codeName":"SSLHandshakeFailed","errmsg":"no SSL certificate provided by peer; connection rejected"},"remote":"172.27.0.7:41958","connectionId":4}}
{"t":{"$date":"2024-02-28T23:00:28.316+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn4","msg":"Connection ended","attr":{"remote":"172.27.0.7:41958","connectionId":4,"connectionCount":0}}
{"t":{"$date":"2024-02-28T23:00:30.917+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"172.27.0.7:42100","connectionId":5,"connectionCount":1}}
{"t":{"$date":"2024-02-28T23:00:30.923+00:00"},"s":"E",  "c":"NETWORK",  "id":23255,   "ctx":"conn5","msg":"No SSL certificate provided by peer; connection rejected"}
{"t":{"$date":"2024-02-28T23:00:30.923+00:00"},"s":"I",  "c":"NETWORK",  "id":22988,   "ctx":"conn5","msg":"Error receiving request from client. Ending connection from remote","attr":{"error":{"code":141,"codeName":"SSLHandshakeFailed","errmsg":"no SSL certificate provided by peer; connection rejected"},"remote":"172.27.0.7:42100","connectionId":5}}
{"t":{"$date":"2024-02-28T23:00:30.923+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn5","msg":"Connection ended","attr":{"remote":"172.27.0.7:42100","connectionId":5,"connectionCount":0}}
{"t":{"$date":"2024-02-28T23:00:33.446+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"172.27.0.7:42114","connectionId":6,"connectionCount":1}}
{"t":{"$date":"2024-02-28T23:00:33.452+00:00"},"s":"E",  "c":"NETWORK",  "id":23255,   "ctx":"conn6","msg":"No SSL certificate provided by peer; connection rejected"}
{"t":{"$date":"2024-02-28T23:00:33.452+00:00"},"s":"I",  "c":"NETWORK",  "id":22988,   "ctx":"conn6","msg":"Error receiving request from client. Ending connection from remote","attr":{"error":{"code":141,"codeName":"SSLHandshakeFailed","errmsg":"no SSL certificate provided by peer; connection rejected"},"remote":"172.27.0.7:42114","connectionId":6}}
{"t":{"$date":"2024-02-28T23:00:33.452+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn6","msg":"Connection ended","attr":{"remote":"172.27.0.7:42114","connectionId":6,"connectionCount":0}}
{"t":{"$date":"2024-02-28T23:00:36.021+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"172.27.0.7:42128","connectionId":7,"connectionCount":1}}
{"t":{"$date":"2024-02-28T23:00:36.027+00:00"},"s":"E",  "c":"NETWORK",  "id":23255,   "ctx":"conn7","msg":"No SSL certificate provided by peer; connection rejected"}
{"t":{"$date":"2024-02-28T23:00:36.027+00:00"},"s":"I",  "c":"NETWORK",  "id":22988,   "ctx":"conn7","msg":"Error receiving request from client. Ending connection from remote","attr":{"error":{"code":141,"codeName":"SSLHandshakeFailed","errmsg":"no SSL certificate provided by peer; connection rejected"},"remote":"172.27.0.7:42128","connectionId":7}}
{"t":{"$date":"2024-02-28T23:00:36.027+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn7","msg":"Connection ended","attr":{"remote":"172.27.0.7:42128","connectionId":7,"connectionCount":0}}
{"t":{"$date":"2024-02-28T23:00:38.628+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"172.27.0.7:42130","connectionId":8,"connectionCount":1}}
{"t":{"$date":"2024-02-28T23:00:38.634+00:00"},"s":"E",  "c":"NETWORK",  "id":23255,   "ctx":"conn8","msg":"No SSL certificate provided by peer; connection rejected"}
{"t":{"$date":"2024-02-28T23:00:38.634+00:00"},"s":"I",  "c":"NETWORK",  "id":22988,   "ctx":"conn8","msg":"Error receiving request from client. Ending connection from remote","attr":{"error":{"code":141,"codeName":"SSLHandshakeFailed","errmsg":"no SSL certificate provided by peer; connection rejected"},"remote":"172.27.0.7:42130","connectionId":8}}
{"t":{"$date":"2024-02-28T23:00:38.634+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn8","msg":"Connection ended","attr":{"remote":"172.27.0.7:42130","connectionId":8,"connectionCount":0}}
{"t":{"$date":"2024-02-28T23:00:41.176+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"172.27.0.7:43258","connectionId":9,"connectionCount":1}}
{"t":{"$date":"2024-02-28T23:00:41.182+00:00"},"s":"E",  "c":"NETWORK",  "id":23255,   "ctx":"conn9","msg":"No SSL certificate provided by peer; connection rejected"}
{"t":{"$date":"2024-02-28T23:00:41.182+00:00"},"s":"I",  "c":"NETWORK",  "id":22988,   "ctx":"conn9","msg":"Error receiving request from client. Ending connection from remote","attr":{"error":{"code":141,"codeName":"SSLHandshakeFailed","errmsg":"no SSL certificate provided by peer; connection rejected"},"remote":"172.27.0.7:43258","connectionId":9}}
{"t":{"$date":"2024-02-28T23:00:41.182+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn9","msg":"Connection ended","attr":{"remote":"172.27.0.7:43258","connectionId":9,"connectionCount":0}}
++ eval 'mongo_eval '\''quit(db.stats().ok ? 0 : 1);'\'''
+++ mongo_eval 'quit(db.stats().ok ? 0 : 1);'
+++ local 'eval=quit(db.stats().ok ? 0 : 1);'
+++ shift
+++ mongo --quiet --eval 'function count(coll) { return coll.count() }; function upsert(coll, doc) { return coll.save(doc) }; quit(db.stats().ok ? 0 : 1);'
+++ docker run --rm -i --cap-add SYS_NICE --link mongo-container-15608-30175:mongo --entrypoint mongo librarytest/mongo-tls:mongo-4.4.29 --host mongo --tls --tlsCAFile /certs/ca.crt --quiet --eval 'function count(coll) { return coll.count() }; function upsert(coll, doc) { return coll.save(doc) }; quit(db.stats().ok ? 0 : 1);'
Error: network error while attempting to run command 'isMaster' on host 'mongo:27017'  :
connect@src/mongo/shell/mongo.js:374:17
@(connect):3:6
exception: connect failed
exiting with code 1
failed
```

**Comment by tianon** [comment]
Got it:

```console
$ ./test/run.sh -t mongo-tls-basics -t mongo-tls-auth mongo:4.4.29
testing mongo:4.4.29
	'mongo-tls-basics' [1/2]....passed
	'mongo-tls-auth' [2/2]....passed
```

(However, I should probably build all of 4.4.29, 5.0.25, 6.0.14, 7.0.6, 7.0.7-rc0 and verify each new release because they all got this change included/backported.)

**Comment by tianon** [comment]
```console
$ ./test/run.sh -t mongo-tls-basics -t mongo-tls-auth mongo:4.4 mongo:5.0 mongo:6.0 mongo:7.0 mongo:7.0-rc
testing mongo:4.4
	'mongo-tls-basics' [1/2]....passed
	'mongo-tls-auth' [2/2]....passed
testing mongo:5.0
	'mongo-tls-basics' [1/2]...passed
	'mongo-tls-auth' [2/2]....passed
testing mongo:6.0
	'mongo-tls-basics' [1/2]...passed
	'mongo-tls-auth' [2/2].....passed
testing mongo:7.0
	'mongo-tls-basics' [1/2]...passed
	'mongo-tls-auth' [2/2].....passed
testing mongo:7.0-rc
	'mongo-tls-basics' [1/2]...passed
	'mongo-tls-auth' [2/2].....passed
```

File: test/tests/mongo-basics/run.sh [hunk 1]
```diff
@@ -55,35 +55,24 @@ if [[ "$testName" == *tls* ]]; then
...
 		COPY --from=certs --chown=mongodb:0 /certs /certs
 		RUN cat /certs/cert.crt /certs/private.key > /certs/both.pem # yeah, what
 	EOD
 	image="$tlsImage"
 	mongodRunArgs+=(
 		--hostname mongo
 	)
-	# test for 4.2+ (where "s/ssl/tls/" was applied to all related options/flags)
-	# see https://docs.mongodb.com/manual/tutorial/configure-ssl/#procedures-using-net-ssl-settings
-	if docker run --rm "$image" mongod --help 2>&1 | grep -q -- ' --tlsMode '; then
```
Comment (yosifkit) [inline test/tests/mongo-basics/run.sh:68]: 👀 do we still need the check for `percona:psmdb-4.2`

Comment (tianon) [reply test/tests/mongo-basics/run.sh:68]:
```console
$ ./test/run.sh -t mongo-tls-basics percona:psmdb-6.0.6 percona:psmdb-5.0.18 percona:psmdb-4.4.22 percona:psmdb-4.2.24
testing percona:psmdb-6.0.6
        'mongo-tls-basics' [1/1]...passed
testing percona:psmdb-5.0.18
        'mongo-tls-basics' [1/1]....passed
testing percona:psmdb-4.4.22
        'mongo-tls-basics' [1/1]....passed
testing percona:psmdb-4.2.24
        'mongo-tls-basics' [1/1]....passed
```

:thinking: looks like we're fine?

Comment (tianon) [reply test/tests/mongo-basics/run.sh:68]: Oh, because 4.2+ switched, so we are fine. :smile:

**Review by yosifkit — APPROVED** [review]

**Pull request merged**